whatshap/cli/extend.py

```diff
@@ -0,0 +1,206 @@
+        help="Reference file. Must be accompanied by .fai index (create with samtools faidx)")
+    arg("--gap-threshold", "-g", metavar="GAPTHRESHOLD", default=70, type=int,
+        help="Threshold percentage of qualities for assigning phase information to a variant.")
+    arg("--cut-poly", "-c", metavar="CUTPOLY", default=10, type=int,
+        help="ignore polymers longer than the cut value.")
+    arg("--only-indels", "-i", default=False, action="store_true",
+        help="extend new phasing information only to indels.")
+    arg("--ignore-read-groups", default=False, action="store_true",
+        help="Ignore read groups in BAM/CRAM header and assume all reads come from the same sample.")
+    arg("--chromosome", dest="chromosomes", metavar="CHROMOSOME", default=[], action="append",
+        help="Name of chromosome to phase. If not given, all chromosomes in the input VCF are phased. "
+        "Can be used multiple times.")
+    arg("variant_file", metavar="VCF", help="VCF file with phased variants (must be gzip-compressed and indexed)")
+    arg("alignment_file", metavar="ALIGNMENTS",
+        help="BAM/CRAM file with alignments to be tagged by haplotype")
+
+
+# fmt: on
+
+
+def run_extend(
+    variant_file,
+    alignment_file,
+    output=None,
+    reference: Union[None, bool, str] = False,
+    ignore_read_groups: bool = False,
+    only_indels: bool = False,
+    chromosomes: Optional[List[str]] = None,
+    gap_threshold: int = 70,
+    cut_poly: int = -1,
+    write_command_line_header: bool = True,
+    tag: str = "PS",
+):
+    timers = StageTimer()
+    timers.start("extend-run")
+    command_line: Optional[str]
+    if write_command_line_header:
+        command_line = "(whatshap {}) {}".format(__version__, " ".join(sys.argv[1:]))
+    else:
+        command_line = None
+    with ExitStack() as stack:
+        logger.debug("Creating PhasedInputReader")
+        phased_input_reader = stack.enter_context(
+            PhasedInputReader(
+                [alignment_file],
+                None if reference is False else reference,
+                NumericSampleIds(),
+                ignore_read_groups,
+                only_snvs=False,
+            )
+        )
+        logger.debug("Creating PhasedVcfWriter")
+        try:
+            vcf_writer = stack.enter_context(
+                PhasedVcfWriter(
+                    command_line=command_line,
+                    in_path=variant_file,
+                    out_file=output,
+                    tag=tag,
+                )
+            )
+        except (OSError, VcfError) as e:
+            raise CommandLineError(e)
+
+        vcf_reader = stack.enter_context(VcfReader(variant_file, phases=True))
+
+        try:
+            bam_reader = stack.enter_context(
+                pysam.AlignmentFile(
+                    alignment_file,
+                    reference_filename=reference if reference else None,
+                    require_index=True,
```

Comment:
If this is kept, maybe add a command line arg threads and pass that on to the pysam threads argument to allow for increased decompression speed.